src/main/java/org/cloudfoundry/community/servicebroker/s3/service/iam/Iam.java

```diff
@@ -0,0 +1,142 @@
+package org.cloudfoundry.community.servicebroker.s3.service.iam;
+
+import com.amazonaws.services.identitymanagement.AmazonIdentityManagement;
+import com.amazonaws.services.identitymanagement.model.*;
+import org.cloudfoundry.community.servicebroker.s3.policy.BucketGroupPolicy;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+
+public abstract class Iam {
```

Comment:
Can you explain this approach?  It seems like `Iam` needs to be an interface containing some of the methods in this abstract class along with all the methods in `BasicPlanIam`.  No other classes should be reference `BasicPlanIam` but rather only the `Iam` interface. Otherwise it seems like `BasicPlan` and `BasicPlanIam` are very tightly coupled and shouldn't be in separate packages because they would never be used independently (basically the Common Closure and Common Reuse principles).